partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rex/ExprError.kt

```diff
@@ -0,0 +1,12 @@
+package org.partiql.eval.internal.operator.rex
+
+import org.partiql.eval.Environment
+import org.partiql.eval.ExprValue
+import org.partiql.spi.errors.TypeCheckException
+import org.partiql.spi.value.Datum
+
+internal class ExprError : ExprValue {
+    override fun eval(env: Environment): Datum {
+        throw TypeCheckException("A warning has been converted into an error.")
```

Comment:
I was testing the CLI using the example provided in https://github.com/partiql/partiql-lang-kotlin/issues/1680. Works as expected:

```
-- running in strict mode
partiql ▶ 'foo' || 123
   | 
w: [semantic] Undefined function "﷯concat"(STRING, INTEGER). Did you mean: 
- ﷯concat(lhs: CHAR(256), rhs: CHAR(256))
- ﷯concat(lhs: VARCHAR(256), rhs: VARCHAR(256))
- ﷯concat(lhs: STRING, rhs: STRING)
- ﷯concat(lhs: CLOB(2147483647), rhs: CLOB(2147483647))

org.partiql.spi.errors.TypeCheckException: A warning has been converted into an error.
```

Wonder if we could plumb through the specific warning rather than just saying a warning has been changed to an error.

Reply:
We totally can. I'll tackle this in a follow-up PR. I've added the task to the requirements pre-V1 launch list.